### DISCUSSION TITLE
Is there an http endpoint to resume a workflow in Elsa 3?

### DISCUSSION THREAD

**Discussion by jirimatejka** [elsa-workflows/elsa-core#5963] [Design & tech]
Hello,

am learning Elsa v3 API and am wondering if there is an http endpoint to resume a workflow instance by the instance id and bookmark id? Am able to execute a new workflow via HTTP, but can't find any method in the elsa swagger to resume the workflow instance.

thanks

Jiri

**Comment by gtejeda** [comment]
+1

**Comment by sfmskywalker** [comment]
Short answer: I do not see a generic Swagger endpoint that takes `workflowInstanceId` + `bookmarkId` directly.

In the current codebase, the built-in HTTP resume endpoint is `/bookmarks/resume`, but it resumes from a SAS token that already contains the bookmark and workflow instance IDs rather than raw IDs in the request (`src/modules/Elsa.Workflows.Api/Endpoints/Bookmarks/Resume/Endpoint.cs`, especially the route and token handling in lines 24-35, and the resume request creation in lines 146-174).

If you want to resume by ID programmatically, the supported runtime surface is:

- `IWorkflowResumer.ResumeAsync(new ResumeBookmarkRequest { BookmarkId = ..., WorkflowInstanceId = ..., Input = ... })` (`src/modules/Elsa.Workflows.Runtime/Contracts/IWorkflowResumer.cs:32-38`)
- or the newer client API: `var client = await workflowRuntime.CreateClientAsync(workflowInstanceId); await client.RunInstanceAsync(new RunWorkflowInstanceRequest { BookmarkId = bookmarkId, Input = ... });`

The request model exposes `BookmarkId` directly (`src/modules/Elsa.Workflows.Runtime/Messages/RunWorkflowInstanceRequest.cs:17-20`), and the component tests resume bookmarked instances that way (`test/component/Elsa.Workflows.ComponentTests/Scenarios/Variables/CountdownWorkflowTests.cs:48-54`).

So if the question is specifically “is there a plain REST endpoint for raw instance/bookmark IDs in Swagger?”, I do not see one in the current repo. The stock options are the SAS-token endpoint for externally resumable bookmarks, or using the runtime/client API in your own endpoint/service.
